NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -696,17 +700,30 @@ class NeuroID private constructor(
         return SessionIDOriginResult(origin, originCode, sessionID)
     }
 
-
+    @Synchronized
     fun pauseCollection() {
         isSDKStarted = false
-        CoroutineScope(Dispatchers.IO).launch {
+        pauseCollectionJob = CoroutineScope(Dispatchers.IO).launch {
             nidJobServiceManager.sendEventsNow(NIDLogWrapper(), true)
             nidJobServiceManager.stopJob()
             saveIntegrationHealthEvents()
         }
     }
 
+    @Synchronized
```

Comment:
Just curious, what does this tag signify/do?

Reply:
Just to make sure, this was added to ensure that the method will complete before being called again. `Marks the JVM method generated from the annotated function as synchronized, meaning that the method will be protected from concurrent execution by multiple threads by the monitor of the instance (or, for static methods, the class) on which the method is defined.` from the docs.